discovery-provider/src/queries/get_route_metrics.py

```diff
@@ -1,12 +1,294 @@
+                            'total': total_count_records[timestamp]
+                        })
+                return metrics
+            raise exceptions.ArgumentError("Invalid bucket_size for time_range")
+        if time_range == 'month':
+            if bucket_size == 'day':
+                unique_counts = (
+                    session.query(
+                        DailyUniqueUsersMetrics.timestamp,
+                        DailyUniqueUsersMetrics.count
+                    )
+                    .filter(thirty_days_ago <= DailyUniqueUsersMetrics.timestamp)
+                    .filter(DailyUniqueUsersMetrics.timestamp < today)
+                    .all()
+                )
+                unique_count_records = ft.reduce(lambda acc, curr: \
+                    acc.update({str(curr[0]): curr[1]}) or acc, unique_counts, {})
+
+                total_counts = (
+                    session.query(
+                        DailyTotalUsersMetrics.timestamp,
+                        DailyTotalUsersMetrics.count
+                    )
+                    .filter(thirty_days_ago <= DailyTotalUsersMetrics.timestamp)
+                    .filter(DailyTotalUsersMetrics.timestamp < today)
+                    .all()
+                )
+                total_count_records = ft.reduce(lambda acc, curr: \
+                    acc.update({str(curr[0]): curr[1]}) or acc, total_counts, {})
+
+                metrics = []
+                for timestamp, unique_count in unique_count_records.items():
+                    if timestamp in total_count_records:
+                        metrics.append({
+                            'timestamp': timestamp,
+                            'unique': unique_count,
+                            'total': total_count_records[timestamp]
+                        })
+                return metrics
+            if bucket_size == 'week':
+                unique_counts = (
+                    session.query(
+                        func.date_trunc(bucket_size, DailyUniqueUsersMetrics.timestamp).label('timestamp'),
+                        func.sum(DailyUniqueUsersMetrics.count).label('count')
+                    )
+                    .filter(thirty_days_ago <= DailyUniqueUsersMetrics.timestamp)
+                    .filter(DailyUniqueUsersMetrics.timestamp < today)
+                    .group_by(func.date_trunc(bucket_size, DailyUniqueUsersMetrics.timestamp))
+                    .all()
+                )
+                unique_count_records = ft.reduce(lambda acc, curr: \
+                    acc.update({str(curr[0]): curr[1]}) or acc, unique_counts, {})
+
+                total_counts = (
+                    session.query(
+                        func.date_trunc(bucket_size, DailyTotalUsersMetrics.timestamp).label('timestamp'),
+                        func.sum(DailyTotalUsersMetrics.count).label('count')
+                    )
+                    .filter(thirty_days_ago <= DailyTotalUsersMetrics.timestamp)
```

Comment:
wondering if this query does something funky at the beginning of the thirty days ago --

if today is Feb 26, 30 days ago is Jan 27, would the first date truck week only show partial data?